src/Microsoft.Azure.WebJobs/StorageAccountAttribute.cs

```diff
@@ -18,10 +18,15 @@ public sealed class StorageAccountAttribute : Attribute
         /// <summary>
         /// Constructs a new instance.
         /// </summary>
-        /// <param name="account">The name of the Azure Storage connection string to use. This should be the name
-        /// of one of the connection strings present in the application settings (or environment variables). The
-        /// connection string name in settings should be prefixed with the standard "AzureWebJobs" prefix, but the
-        /// value you specify here should not include that prefix.</param>
+        /// <param name="account">A string value indicating the Azure Storage connection string to use. This
+        /// string should be in one of the following formats. These checks will be applied in order and the
+        /// first match wins.
+        /// - The name of an "AzureWebJobs" prefixed app setting or connection string name. E.g., if your setting
+        ///   name is "AzureWebJobsMyStorage", you can specify "MyStorage" here.
+        /// - Can be a string containing %% values (e.g. %StagingStorage%). The value provided will be passed
+        ///   to any INameResolver registered on the JobHostConfiguration to resolve the actual setting name to use.
+        /// - Can be an app setting or connection string name of your choosing.
+        /// </param>
         public StorageAccountAttribute(string account)
```

Comment:
I'm suggesting that I'd like to support `[DocumentDbConnectionAttribute(string connection)]` and have the same behavior you have above -- looking for `AzureWebJobs`+connection, looking for %connection%, looking for an app setting named that specifically, looking for a connection string named that specifically.

Same thing for NH and EasyTables. I can see all of them using that same behavior. The BindingProviders would still need to call something to do the actual creation of the DocumentClient, MobileServiceClient, etc, but that logic of doing the lookups would be nice to share.